src/main.js

```diff
@@ -55,6 +58,17 @@ window.storefrontApp = {
   }
 }
 
+ecomPassport.once('login', () => {
+  sendCart()
+})
+
+hasCartCreated.then(_id => {
+  const { name, params } = router.currentRoute
+  if (['cart', 'checkout'].includes(name) && params.id !== _id) {
```

Comment:
Bonito seu código aqui, mas eu abriria mão da boniteza pela performance:

```suggestion
  if ((name === 'cart' || name === 'checkout') && params.id !== _id) {
```

Na verdade acho que iria até de switch case, mais verboso mas acho que não chega a atrapalhar porque são poucas opções. Acho que o código também é mais legível quando usamos expressões mais comuns e simples para fazer coisas simples haha

Não vou travar a PR por isto, também não é um problema de performance aqui (claro que não).
A dissertação é mais para pensarmos sobre isto nas próximas vezes também, se discordar, ou se já tiver isto em mente, me truque!